odtp/dashboard/utils/middleware.py

```diff
@@ -0,0 +1,119 @@
+import jwt, logging
+from jwt import PyJWKClient
+from fastapi import Request, status, HTTPException, Cookie, Response
+from starlette.middleware.base import BaseHTTPMiddleware
+from nicegui import Client,app
+from odtp.dashboard.utils.storage import save_to_storage, get_from_storage
+from fastapi.responses import RedirectResponse, Response
+import odtp.mongodb.db as db
+
+
+unrestricted_page_routes = {'/', '/components'}
+
+def jwt_decode_header(encoded:str): 
+    """Decodes the header of a JWT token. (Giving data like algorithm and token type as well as signature key: alg, typ, kid.)"""
+    token_header = jwt.get_unverified_header(encoded)
+    return token_header
```

Comment:
It would be better to remove this from the code as it is not used. Otherwise you might use it by mistake in the future, which would introduce some vulnerabilities.